capsules/core/src/test/capsule_test.rs

```diff
@@ -0,0 +1,54 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+//! Interface for running capsule tests.
+//!
+//! As Tock capsules are asynchronous, it is difficult for a test runner to
+//! determine when a test when a test has finished. This interface provides a
+//! `done()` callback to notify when the test is done.
```

Comment:
This doesn't seem right. The tests should be testing the same flow that the kernel and apps will use.

If it is too difficult to determine if a operation has finished then the HIL should be improved to make it easier. Adding special callbacks that only tests will use will just lead to testing different use cases then we are actually running. Which then isn't really testing anything useful

Reply:
I'm not sure I follow. I updated the tests in with this PR, how would the test runner know when the test is finished? A test might cover multiple HILs, so the test runner cannot (always) use a HIL interface. Also, a test might use one asynchronous operation, or a thousand. Only the test can know when it is done.

Reply:
Ah. I misunderstood. I thought you were wanting to add this to any capsule, not just test capsules